eox_tagging/models.py

```diff
@@ -47,35 +49,55 @@ def invalid(self):
         """Returns all invalid tags."""
         return self.exclude(invalidated_at=None)
 
-    def find_by_owner(self, owner):
+    def find_by_owner(self, owner_type, owner_id):
         """Returns all valid tags owned by owner_id."""
-        model = owner.__class__
-        ctype = ContentType.objects.get_for_model(model)
 
-        return self.valid().filter(
+        owner, ctype = self.__get_object_for_this_type(owner_type, owner_id)
+
+        return self.filter(
             owner_type=ctype,
             owner_object_id=owner.id,
         )
 
-    def find_all_tags_for(self, target_object):
-        """Returns all valid on an object."""
-        if get_model_name(target_object) in OPAQUE_KEY_PROXY_MODEL_TARGETS:
-            model = OpaqueKeyProxyModel
-            target_id = OpaqueKeyProxyModel.objects.get(opaque_key=target_object.course_id).id
-        else:
-            model = target_object.__class__
-            target_id = target_object.id
-        ctype = ContentType.objects.get_for_model(model)
+    def find_all_tags_for(self, target_type, target_id):
+        """Returns all valid tags on an object."""
+        if target_type in OPAQUE_KEY_PROXY_MODEL_TARGETS:
+            target_type = PROXY_MODEL_NAME
 
-        return self.valid().filter(
+        target, ctype = self.__get_object_for_this_type(target_type, target_id)
+
+        return self.filter(
             target_type=ctype,
-            target_object_id=target_id,
+            target_object_id=target.id,
         )
 
     def hard_delete(self):
         """ Method for deleting Tag objects"""
         return super(TagQuerySet, self).delete()
 
+    def __get_object_for_this_type(self, object_type, object_id):
```

Comment:
1. imo this deserves a better doc string, my **new** personal preference http://google.github.io/styleguide/pyguide.html#383-functions-and-methods
2.this deserves more tests, you only test when object_type=user,I reckon __get_object_for_this_type should be its own test
3.I reckon this fails when this returns None in line 99.
4. there is no consistency in the input, sometimes is a string sometimes is a dict, my advice is that the input always  must be a dict with the key 
`object_id = {'username': 'pepito'} for user`
`object_id = {'username': 'pepito', 'course_id': 'course-v1:edX+FUN101x+3T2017'}` for courseenrollment
and you can make a call like this ctype.get_object_for_this_type(**object_id)

5 you repeat multiple times object_instance = ctype.get_object_for_this_type with the previous suggestion you can generalize that if you include user, site, courseenrollment in a constan, something like

```
    if object_type == constant:
        object_instance = ctype.get_object_for_this_type(**object_id)
```
6 the unique case is `object_type == "opaquekeyproxymodel"`,  you could modify the input 
`object_id['opaque_key'] =  CourseKey.from_string(object_id.get('opaque_key'))`

Reply:
Good idea. I changed it, what do you think now?

Reply:
This looks better, however you had multiple conditions in the previous commit , why did you remove those ?
What happen if object_type is "site_configurations" ?

Reply:
I added error handling in case object_type does not exist in contenttypes